.github/workflows/codespell.yml

```diff
@@ -0,0 +1,22 @@
+---
+name: Codespell
+
+on:
+  push:
+    branches: [main]
```

Comment:
I can't think of a reason this needs to run on pushes to `main` or any other branch.  Is there some other consideration for this?

Reply:
it is common for projects to run CI upon changes to be about to be contributed (PRs) or directly pushed (thus on branches, in particular `main`).  This allows to identify which changes introduced the problem (unlike on schedule -- might need to look for range of changes).  If you want some other way - I guess could be changed here.

Reply:
FWIW, I have been quite an abuser of github actions (see mass use and dashboards in e.g. https://github.com/datalad/datalad-extensions or https://github.com/datalad/git-annex )... this workflow is such a quick/tiny one that I would have not worried about it making any notable dent but I could be proven wrong.

> the workflow is either running on a fork, or a PR and skip it for everything else.

to a degree it does that already -- as merge of a PR creates a new commit, which might be different from "perspective" (those always differ from those actually merged) merge commit which github prepared during PR, then it might run it on an identical state to a PR... yeah, not sure how to formulate for github smth like "if you checked already the state in PR perspective merge do not check after  merge if the same state".  may be @jwodder knows?

Reply:
Okay, we can drop my concern then.  Thanks for trying.

Reply:
Returning to this after all this time — as a repo maintainer, my question is “does running this on commits to `main` make any difference at all”?

Realistically I am not going to go manually looking for any workflow results. I think if it doesn’t block work that needs doing, and no-one is specifically looking for failures, the failures, if any, are going to be ignored; so it seems to me that we really don’t need to run the checks at all.

@cevich perhaps it would make sense to run the `codespell` during the daily cirrus-cron CI runs, which we do follow. (Or, alternatively and purely hypothetically, replace the daily cirrus-cron CI runs by whatever it is where the GitHub workflow run on `main` commits shows up — the point is that both should ideally show up in only one shared place.)


---

Details: How is this test ever going to break if it wasn’t broken on a PR?
- It could be broken in a PR and manually merged. Too bad, that needs to be possible at least for repo admins. We would find out on the next PR.
- It could be something committed without a PR. Maybe that might not need to be possible, the branch can be configured to require PRs. But then again, repo admins could override.
- It seems to me that by far the most likely way for this test to break is if the dictionary of `codespell` changed (it does change every few days). In that case, on _past release branches_, I think we actively don’t want to change code; so we wouldn’t want to run codespell there, at all. We do want to notice such breakage on `main`.

Reply:
@mtrmac github actions is stupid, it can't tell the difference between main in **this** repo, and a fork of this repo.  There's (unfortunately) a lot of users that will make changes on their main branch which end up in PRs upstream.  Having this run on `main` means they will get feedback BEFORE creating a PR.  Sometimes this is helpful.  Most of the time it confuses the hell out of maintainers (like it just did :wink:)

I'm not concerned about this running on the upstream main.  I think we can just ignore it.  If that turns into a problem, I can easily add an "on failure" notification e-mail to the workflow.

You're catch-in-from-cirrus-cron idea would work as well, but that would necessitate migrating the workflow into the cirrus YAML.  So, slightly more work.